### PR TITLE
feat(TG-373): Implement picture preview functionality

### PR DESCRIPTION
**Pull request by Appius**
Implemented:
TG-373 Implement uploaded picture preview
TG-314 Implement map view markup

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/174" target="_blank">ReviewNinja

File: src/app/constructor/map/map-constructor.component.ts [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
+    constructor(private route: ActivatedRoute, private ngZone: NgZone) { }
 
-    ngOnInit(): void {
+    public ngOnInit(): void {
         this.sensors = this.route.snapshot.data['sensors'];
+
+        let self = this;
+
+        /* workaround, because handler code is executed outside of Angular Zone ('this' references to the wrong object) */
+        this.uploader.onAfterAddingFile = this.ngZone.run(() => function (fileItem: any) {
```
Comment (borys-ihnatyev) [inline src/app/constructor/map/map-constructor.component.ts]: arrow function will save context *this*
```js
this.ngZone.run(() => (fileItem: any) => {
    this.reader.readAsDataURL(fileItem._file);
    this.uploader.remove(fileItem);
});
```

File: src/app/constructor/map/map-constructor.component.ts [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+            self.reader.readAsDataURL(fileItem._file);
+            (this as FileUploader).removeFromQueue(fileItem);
+        });
+
+        this.reader.onload = this.ngZone.run(() => function (event: any) {
```
Comment (borys-ihnatyev) [inline src/app/constructor/map/map-constructor.component.ts]: use arrow function to save context
```js
this.ngZone.run(() => (event: any) => {
    this.picture = event.target.result;
});
```

File: src/app/constructor/map/map-constructor.component.ts (hunk 1 above)
Comment (Appius) [reply src/app/constructor/map/map-constructor.component.ts]: !fixed thanks!

File: src/app/constructor/map/map-constructor.component.ts (hunk 2 above)
Comment (Appius) [reply src/app/constructor/map/map-constructor.component.ts]: !fixed thanks!

File: src/app/constructor/map/map-constructor.style.scss [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+.inputs {
+    &__item {
+        width: 48%;
+        float: left;
+        margin-right: 4%;
+        &:last-child {
+            margin-right: 0;
+        }
+    }
+    .label {
```
Comment (piton13) [inline src/app/constructor/map/map-constructor.style.scss:28]: will be better to use empty line.

File: src/app/constructor/map/map-constructor.style.scss [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+.switch {
+    margin-top: 20px;
+    float: right;
+
+    &__checkbox {
+        height: 15px;
+    }
+    &__label {
```
Comment (piton13) [inline src/app/constructor/map/map-constructor.style.scss:50]: the same.

File: src/app/constructor/map/map-constructor.style.scss [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+.nv-file-over {
+    border: dotted 3px red;
+}
+
+.add-picture {
+    width: 150px;
+    margin: 120px auto;
+    &__icon {
```
Comment (piton13) [inline src/app/constructor/map/map-constructor.style.scss:73]: also.

File: src/app/constructor/map/map-constructor.style.scss [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        width: 60px;
+        height: 60px;
+        margin: 0 auto;
+        background-image: url('assets/ic-add-a-photo.svg');
+        background-repeat: no-repeat;
+    }
+    &__title {
```
Comment (piton13) [inline src/app/constructor/map/map-constructor.style.scss:81]: and here one more.

File: src/app/constructor/map/map-constructor.style.scss (hunk 3 above)
Comment (Appius) [inline src/app/constructor/map/map-constructor.style.scss:28]: !fixed

File: src/app/constructor/map/map-constructor.style.scss (hunk 4 above)
Comment (Appius) [inline src/app/constructor/map/map-constructor.style.scss:50]: !fixed

File: src/app/constructor/map/map-constructor.style.scss (hunk 5 above)
Comment (Appius) [inline src/app/constructor/map/map-constructor.style.scss:73]: !fixed

File: src/app/constructor/map/map-constructor.style.scss (hunk 6 above)
Comment (Appius) [inline src/app/constructor/map/map-constructor.style.scss:81]: !fixed

**Pull request merged**